hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java

```diff
@@ -63,13 +63,12 @@ public HederaSelfDestructOperation(
 	@Override
 	public OperationResult execute(final MessageFrame frame, final EVM evm) {
 		final var updater = (HederaStackedWorldStateUpdater) frame.getWorldUpdater();
-		final var beneficiaryAddressOrAlias = Words.toAddress(frame.getStackItem(0));
-		final var beneficiaryAddress = updater.priorityAddress(beneficiaryAddressOrAlias);
-		if (!addressValidator.test(beneficiaryAddress, frame)) {
-			return reversionWith(null, HederaExceptionalHaltReason.INVALID_SOLIDITY_ADDRESS);
-		}
+		final var beneficiaryAddress = Words.toAddress(frame.getStackItem(0));
 		final var toBeDeleted = frame.getRecipientAddress();
 		final var beneficiary = updater.get(beneficiaryAddress);
+		if (!addressValidator.test(beneficiaryAddress, frame) || !toBeDeleted.equals(updater.priorityAddress(toBeDeleted))) {
```

Comment:
I don't think the `!toBeDeleted.equals(updater.priorityAddress(toBeDeleted))` check is needed, because if it was `false`, that would mean when the EVM called `get(toBeDeleted)` to create this `MessageFrame`, it would have got `null`.